### DISCUSSION TITLE
Passing form errors **and** data from POST endpoint in the new system

### DISCUSSION THREAD

**Discussion by cdcarson** [sveltejs/kit#5802] [Q&A]
This is a question about [this section in the migration guide](https://github.com/sveltejs/kit/discussions/5774#discussioncomment-3292699), having to do with `+page.server.js`. The guide says:

> POST, PUT and PATCH handlers should not return data. They can return an errors object of validation errors that can be used for UI (provided to the component as an errors prop)...

In this case (form validation errors occurring) it's unclear to me how the originally posted form data gets put back onto the page. For example:

- The user posts `{name: '$#!@%^$', breed: 'yellow'}` to `/kittens/add`
- The `POST` function returns `{errors: {name: 'Kitten name must not contain bad words.'}}`

What happens now? Am I expected to have a`GET` function that (re-)reads the response body and sticks that into `data`? Does SvelteKit do this for me?

Thanks.

**Comment by chvanam** [comment]
If I'm not mistaken the idea is that the GET handler will run before the POST handler when a POST is submitted, where you can get your data as you would normally.
